web/packages/teleport/src/Navigation/SideNavigation/Search.tsx

```diff
@@ -0,0 +1,198 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState, useRef, useEffect } from 'react';
+import { NavLink } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { Box, Flex, Text } from 'design';
+import { height, space, color } from 'design/system';
+
+import { NavigationSection, NavigationSubsection } from './Navigation';
+import {
+  Section,
+  RightPanel,
+  verticalPadding,
+  getSubsectionStyles,
+} from './Section';
+import { CategoryIcon } from './CategoryIcon';
+import { CustomNavigationCategory } from './categories';
+
+export function SearchSection({
+  navigationSections,
+  expandedSection,
+  previousExpandedSection,
+  handleSetExpandedSection,
+  currentView,
+}: {
+  navigationSections: NavigationSection[];
+  expandedSection: NavigationSection;
+  previousExpandedSection: NavigationSection;
+  currentView: NavigationSubsection;
+  handleSetExpandedSection: (section: NavigationSection) => void;
+}) {
+  const section: NavigationSection = {
+    category: CustomNavigationCategory.Search,
+    subsections: [],
+  };
+
+  const isExpanded =
+    expandedSection?.category === CustomNavigationCategory.Search;
+  return (
+    <Section
+      key="search"
+      section={section}
+      $active={false}
+      onClick={() => null}
+      setExpandedSection={() => handleSetExpandedSection(section)}
+      aria-controls={`panel-${expandedSection?.category}`}
+      isExpanded={isExpanded}
+    >
+      <RightPanel
+        isVisible={isExpanded}
+        skipAnimation={!!previousExpandedSection}
+        id={`panel-${section.category}`}
+        onFocus={() => handleSetExpandedSection(section)}
+      >
+        <SearchContent
+          navigationSections={navigationSections}
+          isVisible={isExpanded}
+          currentView={currentView}
+        />
+      </RightPanel>
+    </Section>
+  );
+}
+
+function SearchContent({
+  navigationSections,
+  isVisible,
+  currentView,
+}: {
+  navigationSections: NavigationSection[];
+  isVisible: boolean;
+  currentView: NavigationSubsection;
+}) {
+  const [searchInput, setSearchInput] = useState('');
+  const inputRef = useRef<HTMLInputElement>();
+
+  useEffect(() => {
+    if (isVisible) {
+      setSearchInput('');
+    }
+  }, [isVisible]);
```

Comment:
This may be leftover from some previous iteration, but it doesn't look like this is needed anymore. We can remove it from the props and this useEffect